### DISCUSSION TITLE
How OpenAI filters requests made by bots/scrapers

### DISCUSSION THREAD

**Discussion by rawandahmad698** [rawandahmad698/PyChatGPT#103] [Announcements]
When a request is made to an OpenAI API, the first thing that happens is that a request is made to the **moderation** endpoint. This endpoint is used to process a text for a 'flagged' check, and a 'blocked check', and if the response from the next API request will contain potentially harmful or sensitive information. If the request is determined to be legitimate, and the response is safe, and OpenAI will return an appropriate response.

However, if no request is made to the moderation endpoint, OpenAI will return an unfiltered response. This means that the response will not have been checked for potentially harmful or sensitive information. Because of this, it is important for everyone to always make a request to the moderation endpoint before making any other requests.

Overall, the moderation endpoint is an important tool that OpenAI uses to protect users from potentially harmful or sensitive information that may be contained in unfiltered responses from their APIs. By making a request to the moderation endpoint before making any other requests, users of OpenAI APIs can help ensure the security and integrity of their data.

First: 
<img width="688" alt="image" src="https://user-images.githubusercontent.com/19406683/206899208-eb5478c1-c40b-436b-a14b-558a2f62f0e2.png">

```curl
curl 'https://chat.openai.com/backend-api/moderations' \
-X 'POST' \
-H 'Content-Type: application/json' \
-H 'Accept: */*' \
-H 'Authorization: Bearer token' \
-H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
-H 'Accept-Encoding: gzip, deflate, br' \
-H 'Host: chat.openai.com' \
-H 'Origin: https://chat.openai.com' \
-H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.1 Safari/605.1.15' \
-H 'Connection: keep-alive' \
-H 'Referer: https://chat.openai.com/chat' \
-H 'Content-Length: 11871' \
--data-binary $'{"input":"Hello","model":"text-moderation-playground"}'
```

Then to: 

<img width="615" alt="image" src="https://user-images.githubusercontent.com/19406683/206899293-7115c78f-c1fc-4f7b-9abc-64b1ebb38cde.png">

**Comment by acheong08** [comment]
I am unable to get consistent results. The response is short around 70% of the time I ask with moderation off and 65% of the time with it on. The difference could be completely due to chance.

@transitive-bullshit Do you mind doing some testing to see if this is true? 

P.S. The text here looks like it's from ChatGPT

**Reply by transitive-bullshit** [reply]
I'll follow-up w/ some additional testing for sure; it won't be until tomorrow, though.

**Reply by transitive-bullshit** [reply]
Rawand mentioned he has some larger automated test results that might help.

**Reply by rawandahmad698** [reply]
It is. I generated this using ChatGPT. Who writes stuff now 😂

**Reply by acheong08** [reply]
> It is. I generated this using ChatGPT. Who writes stuff now joy

I feel dumb. Most of my code/docs/readme is still hand written. I didn't bother letting ChatGPT write it for me.

**Reply by rawandahmad698** [reply]
I didn't mean the code. I meant the paragraphs

**Reply by acheong08** [reply]
> I didn't mean the code. I meant the paragraphs

Yeah I know. But who writes stuff anymore, including code... I should start learning how to use ChatGPT.
